pipeline/ephys.py

```diff
@@ -186,3 +197,30 @@ class TrialSpikes(dj.Computed):
     ---
     spike_times : longblob # (s) spike times for each trial, relative to go cue
     """
+
+
+@schema
+class UnitStat(dj.Computed):
+    definition = """
+    -> Unit
+    ---
+    isi_violation: float    # 
+    avg_firing_rate: float  # (Hz)
+    """
+
+    isi_violation_thresh = 0.002  # violation threshold of 2 ms
+
+    # TODO: we need all TrialSpikes to be inserted prior to this calc, thus there may be a race condition here
+
+    key_source = ProbeInsertion & TrialSpikes
```

Comment:
If this is the case, do something like `ProbeInsertion & experiment.SessionTrial - (experiment.SessionTrial - TrialSpikes)`

Reply:
I don't quite get how this would solve the race-condition - when TrialSpikes are being inserted, some inserted but not yet all. If `UnitStat.populate()` is called in the middle if this insertion process, the corresponding calculation will be wrong